src/Executor/Executor.php

```diff
@@ -1148,7 +1148,7 @@ private function defaultTypeResolver($value, $context, ResolveInfo $info, Abstra
     {
         // First, look for `__typename`.
         if ($value !== null &&
-            is_array($value) &&
+            (is_array($value) || $value instanceof \ArrayAccess) &&
```

Comment:
Should also test if is an object since `instanceof` will throw an php error (PHP Fatal error:  instanceof expects an object instance) if $value is not an object. `(is_array($value) || (is_object($value) && $value instanceof ArrayAccess))`

Reply:
@mcg-web I don't think it's the case. It only throws this error when you check against constant (like `null instanceof ArrayAccess`).

Reply:
```
08:07 $ php -a
Interactive mode enabled

php > var_dump('' instanceof \stdClass);
PHP Fatal error:  instanceof expects an object instance, constant given in php shell code on line 1
php > var_dump(1 instanceof \stdClass);
PHP Fatal error:  instanceof expects an object instance, constant given in php shell code on line 1
php > var_dump((new \stdClass) instanceof \stdClass);
php shell code:1:
bool(true)
```

That's what I have when I test this, but maybe I'm just doing something wrong.

Reply:
You are trying inline values, but if you try any variable, it will work

Reply:
You right `$v = 1; var_dump($v instanceof \stdClass);` this work, inline has a different behavior.